test/unittests/filter_iterator_test.cpp

```diff
@@ -0,0 +1,107 @@
+// EVMC: Ethereum Client-VM Connector API.
+// Copyright 2022 The EVMC Authors.
+// Licensed under the Apache License, Version 2.0.
+
+#include <tools/evmc/filter_iterator.hpp>
+#include <gtest/gtest.h>
+#include <cctype>
+
+using evmc::skip_space_iterator;
+
+namespace
+{
+std::string remove_space(std::string_view in)
+{
+    // Copy input to additional buffer. This helps with out-of-buffer reads detection by sanitizers.
+    const std::vector<char> in_buffer(in.begin(), in.end());
+
+    // Filter the input.
+    std::string out;
+    std::copy(skip_space_iterator{in_buffer.begin(), in_buffer.end()},
+              skip_space_iterator{in_buffer.end(), in_buffer.end()}, std::back_inserter(out));
```

Comment:
What is the point of `end,end`?

Reply:
For filter iterator you need iterator pair so for the end filter iterator double end is provided. I considered some other options (default argument value, sentinel) but `boost::filter_iterator` does the same so I left it as is.

Reply:
But couldn't you just pass `end()` as the iterator?

Reply:
```
std::copy(skip_space_iterator{in_buffer.begin(), in_buffer.end()}, in_buffer.end(), std::back_inserter(out));
```

Isn't `end` an iterator itself?

Reply:
Yes, this is also "workable" but the iterators would have different types.
1. Additional overloads for comparison operators are needed (so we can compare `filter_iterator` with `BaseIterator`).
2. The `std::copy()` and many algorithms cannot be used because they expect `InputIterator`s to be of the same type. https://en.cppreference.com/w/cpp/algorithm/copy.

I expect much better user experience with `std::range`, but I also have 0 experience with these so far.

Reply:
Alright, I was wondering as it looked weird. Fine this way to avoid more code.